game/ui/components/agentassignment.cpp

```diff
@@ -84,7 +84,8 @@ void AgentAssignment::init(sp<Form> form, Vec2<int> location, Vec2<int> size)
 			this->currentAgent = agent;
 		}
 		auto icon = c->findControl(ControlGenerator::AGENT_ICON_NAME);
-		if (agent && icon && c->isPointInsideControlBounds(e, icon))
+		if (agent && icon && c->isPointInsideControlBounds(e, icon) &&
+		    Event::isPressed(e->forms().MouseInfo.Button, Event::MouseButton::Right))
```

Comment:
It is not vanilla behaviour. In vanilla X-Com work both mouse buttons for load the equip screen.

Reply:
In vanilla I can drag-&-drop units when clicking on their portraits. Here in OpenApoc I can't do that. When I start dragging - equipment screen pops up. This is why I changed buttons.

Reply:
In this case need to change the drag&drop behaviour. I.e. check the MouseUp event instead the MouseClick on the icon.

Reply:
Ok.

Reply:
I just tested vanilla X-Com behaviour. The equip screen is loaded after the MouseUp event.
I propose the algorithm:

- the MouseDown event for an agentControl or vehicleControl
```
if (mouseOverIcon) {
    remember the iconControl;
}
else {
    remember nullptr;
}
do a conventional drag&drop action for MouseDown; // if needed
```
- the MouseUp event for an agentControl or vehicleControl
```
if (mouseOverIcon && itsTheSameIcon) {
    cancel drag&drop;
    start an equip screen;
}
else {
    do a conventional drag&drop action for MouseUp; // if should be of course
}
```

Reply:
This algorythm won't work because player can start dragging and release button while staying on the same icon. But don't worry. I've made things right already.